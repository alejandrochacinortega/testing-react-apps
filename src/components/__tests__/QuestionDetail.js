import QuestionDetail, { mapStateToProps, QuestionDetailDisplay } from "../QuestionDetail";
import renderer from 'react-test-renderer';
import React from 'react';


import { exportAllDeclaration } from "@babel/types";
describe(`The Question Detail Component`, ()=> {
    describe(`The Container Element`, ()=> {
        describe(`mapStateToProps`, ()=> {
            it("should map the state to props correctly", () => {
                const sampleQuestion = {
                    question_id: 42,
                    body: "Space is big",
                }
                const appState = {
                    questions: [sampleQuestion]
                };
                const ownProps = {
                    question_id: 42
                };  
                const componentState = mapStateToProps(appState, ownProps);
                expect(componentState).toEqual(sampleQuestion)
            })
        })
    })

    describe(`The display element`, () => {
        const tree = renderer.create(
            <QuestionDetailDisplay
                title="The meaning of life"
                body="42"
                answer_count={0}
                tags={["hiking"]}
            ></QuestionDetailDisplay>
        )

        expect(tree.toJSON()).toMatchSnapshot()
    })
    it(`Should not regress`, () => {
        
    })
})