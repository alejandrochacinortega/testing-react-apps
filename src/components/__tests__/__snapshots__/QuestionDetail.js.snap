// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div>
  <h3
    className="mb-2"
  >
    The meaning of life
  </h3>
  <div>
    <div
      className="mb-3"
    >
      <div>
        <code>
          hiking
        </code>
      </div>
    </div>
    <div>
      <p>
        42
      </p>
    </div>
    <div>
      0
       Answers
    </div>
  </div>
</div>
`;
